common/views/components/AudioPlayer/AudioPlayer.tsx

```diff
@@ -131,16 +143,25 @@ type PlayRateProps = {
 };
 
 const PlayRate: FC<PlayRateProps> = ({ audioPlayer, id }) => {
+  const { audioPlaybackRate, setAudioPlaybackRate } = useContext(AppContext);
   const speeds = [0.5, 1, 1.5, 2];
-  const [currentActiveSpeedIndex, setCurrentActiveSpeedIndex] =
-    useState<typeof speeds[number]>(1);
+  const [currentActiveSpeedIndex, setCurrentActiveSpeedIndex] = useState<
+    typeof speeds[number]
+  >(speeds.indexOf(audioPlaybackRate));
+
+  useEffect(() => {
+    setCurrentActiveSpeedIndex(speeds.indexOf(audioPlaybackRate));
```

Comment:
another vague and ill-informed suggestion about making your life easier - is this state definitely necessary? When you do
```typescript
isActive={speeds[currentActiveSpeedIndex] === speed}
```
below, could you not just replace that with
```typescript
isActive={speed === audioPlaybackRate}
```
and do away with `currentActiveSpeedIndex`?

Reply:
I'm pretty sure it definitely isn't necessary. Updated it to be much simpler, thanks.